tools/bin/ci_check_dependency.py

```diff
@@ -81,19 +84,81 @@ def get_depended_connectors(changed_modules, all_build_gradle_files):
     depended_connectors = []
     for changed_module in changed_modules:
         for connector, gradle_file in all_build_gradle_files.items():
+            if gradle_file is None:
+                continue
             with open(gradle_file) as file:
                 if changed_module in file.read():
                     depended_connectors.append(connector)
     return depended_connectors
 
 
+def get_connector_version(connector):
+    with open(f"{CONNECTOR_PATH}/{connector}/Dockerfile") as f:
+        for line in f:
+            if "io.airbyte.version" in line:
+                return line.split("=")[1].strip()
+
+
+def get_connector_version_status(connector, version):
+    if "strict-encrypt" not in connector:
+        return f"`{version}`"
+    if connector == "source-mongodb-strict-encrypt":
+        base_variant_version = get_connector_version("source-mongodb-v2")
+    else:
+        base_variant_version = get_connector_version(connector.replace("-strict-encrypt", ""))
+    if base_variant_version == version:
+        return f"`{version}`"
+    else:
+        return f"❌ `{version}`<br/>(mismatch: `{base_variant_version}`)"
```

Comment:
No suggestion, just wondering why these are handled differently?

Reply:
The `strict-encrypt` and its normal variant should always be on the same version. It happens a lot that people would only bump the version for the normal variant, but forget about the `strict-encrypt` version. So I added a special case for those connectors.

I plan to deprecate all `strict-encrypt` connectors soon. So this should be irrelevant afterwards.

Reply:
Cool, thanks for clarifying :)